mmv1/third_party/terraform/fwprovider/framework_provider_project_test.go

```diff
@@ -0,0 +1,284 @@
+					// set
+					t.Setenv("GOOGLE_CLOUD_PROJECT", "GOOGLE_CLOUD_PROJECT") // used
+					t.Setenv("GCLOUD_PROJECT", "GOOGLE_CLOUD_PROJECT")
+					t.Setenv("CLOUDSDK_CORE_PROJECT", "CLOUDSDK_CORE_PROJECT")
+				},
+				Config: testAccFwProvider_projectInEnvsOnly(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_plugin_framework.default", "project", "GOOGLE_CLOUD_PROJECT"),
+				),
+			},
+			{
+				// GCLOUD_PROJECT is used 3rd if set
+				PreConfig: func() {
+					// unset
+					t.Setenv("GOOGLE_PROJECT", "")
+					t.Setenv("GOOGLE_CLOUD_PROJECT", "")
+					// set
+					t.Setenv("GCLOUD_PROJECT", "GCLOUD_PROJECT") //used
+					t.Setenv("CLOUDSDK_CORE_PROJECT", "CLOUDSDK_CORE_PROJECT")
+				},
+				Config: testAccFwProvider_projectInEnvsOnly(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_plugin_framework.default", "project", "GCLOUD_PROJECT"),
+				),
+			},
+			{
+				// CLOUDSDK_CORE_PROJECT is used 4th if set
+				PreConfig: func() {
+					// unset
+					t.Setenv("GOOGLE_PROJECT", "")
+					t.Setenv("GOOGLE_CLOUD_PROJECT", "")
+					t.Setenv("GCLOUD_PROJECT", "")
+					// set
+					t.Setenv("CLOUDSDK_CORE_PROJECT", "CLOUDSDK_CORE_PROJECT")
+				},
+				Config: testAccFwProvider_projectInEnvsOnly(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_plugin_framework.default", "project", "CLOUDSDK_CORE_PROJECT"),
+				),
+			},
+		},
+	})
+}
+
+func testAccFwProvider_project_emptyStringValidation(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	context := map[string]interface{}{
+		"project": "",
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccFwProvider_projectInProviderBlock(context),
+				PlanOnly:    true,
+				ExpectError: regexp.MustCompile("expected a non-empty string"),
+			},
+		},
+	})
+}
+
+func testAccFwProvider_project_unknownHandling(t *testing.T) {
+
+	context := map[string]interface{}{
+		"random_suffix":      acctest.RandString(t, 10),
+		"test_step":          "0", // each test step updates this value
+		"org_id":             envvar.GetTestOrgFromEnv(t),
+		"billing_account_id": envvar.GetTestBillingAccountFromEnv(t),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFwProvider_projectUnknownHandling(context, "1"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("data.google_provider_config_plugin_framework.default-1", "project", regexp.MustCompile("tf-test-[0-9a-z]{10}-[0-9]{1,}")),
```

Comment:
The project is the data source looks like using the value tf-test-%{random_suffix}-%{test_step} instead of the environment variable. Is it because the project value is known already?

Reply:
To be honest, I'm confused too. The plugin framework-implemented code [explicitly checks for whether the project value is unknown](https://github.com/GoogleCloudPlatform/magic-modules/blob/477e5c94b4011b01a7f18ccd14fbcecd6d1073eb/mmv1/third_party/terraform/fwtransport/framework_config.go.erb#L151) during the decision to use environment variables, so it would use the ENVs if the value is unknown.

I think, given that we cannot reliably trigger behaviours were the handling of unknown values impacts the outputs of the test, it might be best to remove this test case and instead keep the unit tests around unknown value handling.